token-minter/sources/modules/collection_components.move

```diff
@@ -34,277 +36,180 @@ module minter::collection_components {
+        collection_properties::create_properties(value, value, value, value, value, value, value, value, value)
     }
 
-    public entry fun set_collection_description<T: key>(
-        creator: &signer,
-        collection: Object<T>,
+    public fun set_collection_description(
+        collection_owner: &signer,
+        collection: Object<Collection>,
         description: String,
-    ) acquires CollectionRefs, CollectionProperties {
+    ) acquires CollectionRefs {
         assert!(is_mutable_description(collection), error::permission_denied(EFIELD_NOT_MUTABLE));
-        collection::set_description(option::borrow(&authorized_borrow(collection, creator).mutator_ref), description);
+        collection::set_description(
+            option::borrow(&authorized_borrow_refs(collection, collection_owner).mutator_ref),
+            description,
+        );
     }
 
-    public entry fun set_collection_uri<T: key>(
-        creator: &signer,
-        collection: Object<T>,
+    public fun set_collection_uri(
+        collection_owner: &signer,
+        collection: Object<Collection>,
         uri: String,
-    ) acquires CollectionRefs, CollectionProperties {
+    ) acquires CollectionRefs {
         assert!(is_mutable_uri(collection), error::permission_denied(EFIELD_NOT_MUTABLE));
-        collection::set_uri(option::borrow(&authorized_borrow(collection, creator).mutator_ref), uri);
+        collection::set_uri(option::borrow(&authorized_borrow_refs(collection, collection_owner).mutator_ref), uri);
     }
 
-    public(friend) fun set_collection_royalties<T: key>(
-        creator: &signer,
-        collection: Object<T>,
+    public fun set_collection_royalties(
+        collection_owner: &signer,
+        collection: Object<Collection>,
         royalty: royalty::Royalty,
-    ) acquires CollectionRefs, CollectionProperties {
+    ) acquires CollectionRefs {
         assert!(is_mutable_royalty(collection), error::permission_denied(EFIELD_NOT_MUTABLE));
-        royalty::update(option::borrow(&authorized_borrow(collection, creator).royalty_mutator_ref), royalty);
-    }
-
-    inline fun borrow<T: key>(collection: Object<T>): &CollectionRefs {
-        let collection_address = object::object_address(&collection);
-        assert!(
-            contains_collection_refs(collection_address),
-            error::not_found(ECOLLECTION_REFS_DOES_NOT_EXIST)
+        royalty::update(
+            option::borrow(&authorized_borrow_refs(collection, collection_owner).royalty_mutator_ref),
+            royalty
         );
-        borrow_global<CollectionRefs>(collection_address)
     }
 
-    inline fun authorized_borrow<T: key>(collection: Object<T>, creator: &signer): &CollectionRefs {
-        assert_collection_owner(signer::address_of(creator), collection);
-        borrow(collection)
+    inline fun authorized_borrow_refs<T: key>(collection: Object<T>, collection_owner: &signer): &CollectionRefs {
```

Comment:
does it still need to be type bounded with `<T: key>` even if all its usages are for Object<Collection>?

Reply:
applies to other func signatures as well